### PR TITLE
Add referrer policy tests for loading the various subresources via a srcdoc iframe.

### PR DESCRIPTION
**Pull request by bzbarsky**

### CODE REVIEW

**Comment by bzbarsky** [comment]
@kristijanburnik could you take a look, please?  I wasn't sure what the best way to add this was; whether we want more subresource types or something else.

Note that ideally we would also add tests for resources in srcdoc in srcdoc in normal document, to check that UAs walk up the srcdoc chain....

And we probably also need tests for sandboxed iframes; while that's out of the scope of what I'm trying to do here, it might affect how we approach setting this up.  Adding three more subresource types for every existing type seems a bit weird to me, but right now the "how to load" functions are just keyed off subresource type...

**Comment by bzbarsky** [comment]
Oh, and the CI failures are for path length; we can sort that out once we know whether we're doing a new subresource type or not, since that will affect filenames.

**Comment by bzbarsky** [comment]
@mikewest I'm told @kristijanburnik is no longer active on this stuff, so you own these tests.  Can you take a look at the above questions, please?  And update OWNERS?  ;)

**Comment by mikewest** [comment]
Hooray. I own things. Yes, I'll look at these in the morning and try to remember how all of it works.

**Comment by bzbarsky** [comment]
@mikewest It's been a week.  Ping?

**Comment by hillbrad** [comment]
Looking at this - sorry for the delay -  there are still a number of these tests that use the CSP delivery method, which I believe is no longer current?

https://w3c.github.io/webappsec-referrer-policy/#referrer-policy-delivery

**Comment by bzbarsky** [comment]
I have no idea.  Given my experience with the similarly written mixed-content tests, there are likely all sorts of bugs in the tests themselves, even if we ignore spec changes...

**Comment by mikewest** [comment]
I apologize; I didn't get back to this right away, and then it fell off my radar completely.

I think the approach is reasonable in the short term, but, as you note, adds a lot of tests. Since we should probably be doing `srcdoc` tests (and `iframe@sandbox` tests, etc) for resource types other than images, perhaps it makes sense to extend the general test framework to coalesce these new tests into the existing subresource tests without adding a few hundred new tests.

I don't think it's fair to ask you to do that refactoring after ~4 months, though, so I've uploaded a pass at it to https://github.com/w3c/web-platform-tests/tree/referrer-policy. WDYT?

**Comment by kristijanburnik** [comment]
One does not necessarily need to add a lot of tests. Even a single test can
be generated by a single spec.json rule.

The generating framework allows for suppressing (excluded_tests) or
specifying explicit scenario values (i.e. other than * expansion).

It's up to you if you choose to be more general/specific or prefer
whitelisting to blacklisting. :-)

This is all documented here:
https://github.com/w3c/web-platform-tests/blob/master/referrer-policy/README.md

**Comment by bzbarsky** [comment]
> WDYT?

I have a hard time telling whether it works or not, because I have vague memories of the test harness mutating the server state, such that you couldn't do two loads from a single test.  But maybe that was the mixed-content tests instead...  Have you tested the patch?

**Comment by mikewest** [comment]
> Have you tested the patch?

I have. It works for me locally. I'll turn it into a real PR.

**Comment by foolip** [comment]
This PR has been sitting without activity for over a year. @mikewest, https://github.com/web-platform-tests/wpt/pull/3700 says it addresses this PR, but there are lots of changes in this PR and only small changes in the other?

@bzbarsky, are there things in this PR that still aren't covered in master?

File: referrer-policy/generic/common.js [hunk 1]
```diff
@@ -33,8 +33,25 @@ function appendIframeToBody(url, attributes) {
   return iframe;
 }
 
-function loadImage(src, callback, attributes) {
-  var image = new Image();
+
+function loadImage(src, callback, attributes, isSrcDoc)
```
Comment (jeisinger) [inline referrer-policy/generic/common.js]: nit. other arguments use_this_style, so I'd change this to is_src_doc

File: referrer-policy/generic/common.js [hunk 2]
```diff
@@ -73,14 +90,14 @@ function decodeImageData(rgba) {
   return JSON.parse(string_data);
 }
 
-function decodeImage(url, callback, referrer_policy) {
+function decodeImage(url, callback, referrer_policy, isSrcDoc) {
```
Comment (jeisinger) [inline referrer-policy/generic/common.js]: same here

**Review by jeisinger — CHANGES_REQUESTED** [review]
lgtm with nits

**Comment by bzbarsky** [comment]
> @bzbarsky, are there things in this PR that still aren't covered in master?

@foolip, I don't know.  It's been a long time (even longer since you asked the question, I know) and at this point I'd have to go and reverse-engineer this harness yet again to figure out what's going on.

**Comment by foolip** [comment]
Alright, I guess practically speaking the choice is between taking all of these tests and accepting that there's some or a lot of overlap, or throwing them away to avoid possible overlap. I don't know the existing tests or what's in this PR, so @bzbarsky whatever you prefer I can rubberstamp and merge :)

**Comment by bzbarsky** [comment]
I think we should just throw out this PR at this point.  It's not clear to me that it can even be merged as-is...

**Comment by foolip** [comment]
It had merge conflicts, so it couldn't have been merged without some work. I've closed it now.

**Pull request closed without merging**